File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buy=prices[0];
        int n=prices.size();
        int profit=0;
        for(int i=1;i<n;i++){
            int cost=prices[i]-buy;
            profit=max(profit, cost);
            buy=min(buy, prices[i]);
        }
        return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n=len(prices)
        profit=0
        buy=prices[0]
        for i in range(1, n):
            if prices[i]<buy:
                buy=prices[i]
            profit=max(profit, prices[i]-buy)
        
        return profit